cmd/tink-cli/cmd/hardware/get.go

```diff
@@ -0,0 +1,62 @@
+package hardware
+
+import (
+	"context"
+	"io"
+
+	"github.com/jedib0t/go-pretty/table"
+	"github.com/tinkerbell/tink/client"
+	"github.com/tinkerbell/tink/cmd/tink-cli/cmd/get"
+	"github.com/tinkerbell/tink/protos/hardware"
+)
+
+type GetHardware struct {
+	get.CmdOpt
+	metaClient *client.MetaClient
+}
+
+func NewGetHardwareOpt(metaClient *client.MetaClient) GetHardware {
+	gh := GetHardware{
+		CmdOpt: get.CmdOpt{
+			Headers: []string{"ID", "MAC Address", "IP address", "Hostname"},
+		},
+		metaClient: metaClient,
+	}
+	gh.CmdOpt.PopulateTable = gh.PopulateTable
+	gh.CmdOpt.RetrieveData = gh.RetrieveData
+	return gh
+}
+
+func (h *GetHardware) RetrieveData(ctx context.Context) ([]interface{}, error) {
+	list, err := h.metaClient.HardwareClient.All(ctx, &hardware.Empty{})
+	if err != nil {
+		return nil, err
+	}
+	data := []interface{}{}
+	var hw *hardware.Hardware
+	for hw, err = list.Recv(); err == nil && hw != nil; hw, err = list.Recv() {
+		data = append(data, hw)
+	}
+	if err != nil && err != io.EOF {
+		return nil, err
+	}
+	return data, nil
+}
+
+func (h *GetHardware) PopulateTable(data []interface{}, t table.Writer) error {
+	for _, v := range data {
+		if hw, ok := v.(*hardware.Hardware); ok {
+			// TODO(gianarb): I think we should
+			// print it better. The hardware is one
+			// even if if has more than one
+			// interface.
```

Comment:
agree, maybe we can skip the hw.Id for subsequent nics so it looks like a continuation?

Reply:
Not sure if I understand it but having 2 lines that belong to the same hardware where the first one has an empty ID makes `awk` or `grep` kind of hard. (overall as discussed elsewhere) I think we should not use the table as a format by default but something like `ps`.

Reply:
yeah I don't think we need to support table for programmatic cli use imo. If you want to throw into a pipeline then just use a better machine parse-able format imo.